contracts/PTRC21.sol

```diff
@@ -9,18 +10,148 @@ import "./CopyUtils.sol";
 import "./TRC21.sol";
 import "./Bytes.sol";
 
+contract AbstractTokenTRC21 {
+    function issuer() public view returns (address);
+    function name() public view returns (string memory);
+    function symbol() public view returns (string memory);
+    function decimals() public view returns (uint8);
+}
+
+contract PTRC21 is TRC21 {
+    string private _name;
+    string private _symbol;
+    uint8 private _decimals;
+    uint8 constant PRIVACY_DECIMALS = 8;
+    uint256 private _minFee;
+    uint256 private _depositFee;
+    uint256 private _withdrawFee;
+    address private _token;
+
+    constructor (address token,
+        uint256 minFee,
+        uint256 depositFee,
+        uint256 withdrawFee
+    ) public {
+        if (token != address(0)) {
+            AbstractTokenTRC21 t = AbstractTokenTRC21(token);
+            // require(t.issuer() == msg.sender);
+
+            _token = token;
+            _name =  string(abi.encodePacked("P", t.name()));
+
+            _symbol = t.symbol();
+            _decimals = t.decimals();
+        } else {
+            // Tomo token
+            _token = address(0);
+            _name = "TomoP";
+            _symbol = "TomoP";
+            _decimals = 18;
+        }
+
+        // change the fee correspondent to new decimal in the privacy system
+        (_minFee, _depositFee, _withdrawFee) = _calculatePrivacyFee(minFee, depositFee, withdrawFee);
+        _changeMinFee(minFee);
+        _changeIssuer(msg.sender);
+    }
+
+    /**
+     * @return reduce decimal length.
+     */
+    function _calculatePrivacyFee(uint256 minFee, uint256 depositFee, uint256 withdrawFee) internal view returns (uint256, uint256, uint256) {
+        return (_toPrivacyValue(minFee), _toPrivacyValue(depositFee), _toPrivacyValue(withdrawFee));
+    }
+
+    // convert external value to privacy's
```

Comment:
a bit redundant? as 2 functions basically do the same thing, which can be replaced by a function: convertUnits ...

Reply:
_calculatePrivacyFee will be removed